packages/extension-placeholder/src/placeholder.ts

```diff
@@ -31,15 +31,6 @@ export interface PlaceholderOptions {
       }) => string)
     | string
 
-  /**
-   * **Used for empty check on the document.**
-   *
-   * If true, any node that is not a leaf or atom will be considered for empty check.
-   * If false, only default nodes (paragraphs) will be considered for empty check.
-   * @default false
-   */
-  considerAnyAsEmpty: boolean
```

Comment:
Good point - didn't thought about this. @nperez0111 what do you think? should we push this to a major and make sure to note that the option is gone?

Or should we keep it somehow if possible?

Reply:
Ah sorry I didn't fully document why I think that this was actually OK to remove.

So the reason that was needed was actually masking a fact of the implementation (which was checking if the first child of the doc was the default) with this:
```js
            const isValidNode = this.options.considerAnyAsEmpty
              ? true
              : firstChild && firstChild.type.name === doc.type.contentMatch.defaultType?.name
```

Now, this check is removed because it has become unnecessary in my implementation.

What enabled that?
Previously there was a change made for performance to do a more performant check to see if the editor was empty or not (as opposed to the previous editor.isEmpty): https://github.com/ueberdosis/tiptap/commit/6e5d19f65d98669741973a2812ae13afeea65672

Then we added a more complex check to try to enable placeholders on any node type: https://github.com/ueberdosis/tiptap/pull/4335

But this more complex check did not account for nodes that contain other empty nodes inside of them (issue being `firstChild.nodeSize <= 2` which fails for a node that contains an empty node)

I actually had recently changed the `editor.isEmpty` implementation to be much more performant: https://github.com/ueberdosis/tiptap/commit/f4a7dbaf5c8ab8e6f3bcd17db2cb0802d1f95e22

So, now we could skip all of the complexity around this and go back to the original implementation, but with the nice performance benefit of only check if the editor is empty once.


## TL;DR

That option is no longer necessary anymore. And it is not a breaking change because even if a user specified that value or not, it would not break them and show/hide a placeholder.

Reply:
Unsure about updating docs though. Technically the option is still there in the current release...

One downside of keeping the docs in a separate repo...

Reply:
Oh, thank you for your explanation. That makes so much sense. I didn't realize that `isValidNode` and `considerAnyAsEmpty` are only important for checking the empty state of the document, rather than also being considered for empty nodes in general. But I ran a few tests and everything seems to be fine + re-reading the specific code and your explanation made it clear that it was only for checking the empty document.

Reply:
Yeah re docs: Maybe prepare a PR there and we need to keep track of when it gets released?

Reply:
Hi, @nperez0111! Just came across this PR after some digging that ended up with cloning tiptap and doing `git log v2.3.1..v2.5.7 -S considerAnyAsEmpty` :p

I just have a quick note and question. The note is that, while this is not a breaking change for JS users, it is for TS users, as I just updated tiptap and suddently I came across this error, which I did not understand as I did a minor version bump:

```
Error: Object literal may only specify known properties, and 'considerAnyAsEmpty' does not exist in type 'Partial<PlaceholderOptions>'. (ts)
```

Naturally I scanned the changelog and didn't find anything. The code would still run, but the CI fails, so I'd say it is a bit breaking, not a big problem but confusing nonetheless.

And now for the question: I don't quite remember why I set the thing to be true in the first place, but I'm certain that it didn't work otherwise. So, is the new implementation behaving like the old implementation with `considerAnyAsEmpty = false` or `true`? From my understanding it's the latter, just want to make sure 🙂

Reply:
The considerAnyAsEmpty was a kludge for the isNodeEmpty function not being able to accurately report if content was empty. So not the behavior is as though considerAnyAsEmpty = true but actually catches more cases than just that. But you should try this yourself. 

I could've easily left the type there and mark it deprecated and just totally ignore the value of it. I'll add it back so it doesn't "break" others but it should be removed.

Reply:
Makes sense, thanks for the confirmation

Reply:
I added the deprecated type back with v2.5.9